lib/Partitioner/Partitioner.cpp

```diff
@@ -914,7 +914,40 @@ Partitioner::backendBasedPartition(Function *F,
   for (auto &N : F->getNodes()) {
     for (auto &backend : backends) {
       // Find the first backend that supports this node. The order of backends
-      // is important.
+      // is important. The check flow is :
+
+      // Step 1: If a node is in pre-defined non-supported nodes set, it can not
+      // be assigned to this backend. Goto Step 4.
+
+      // Step 2: If the pre-defined supported nodes set is empty, it means all
+      // nodes could be assigned to this backend. If the pre-defined supported
+      // nodes set is not empty, we check that if the node from Step 1 is in
+      // this set or not. If not, goto Step 4.
+
+      // Step 3: Check if the node is actually supported in this backend, if so,
+      // assign it to this backend and now we find the first backend that
+      // supports this node (The order of backends is important) and break.
+      // Otherwise goto Step 4.
+
+      // Step 4: Continue to check if the node can be
+      // assigned to the next backend.
+
+      // Step 1:
+      auto nonSupported =
```

Comment:
Probably want `const auto &nonSupported = ...` here, or this is going to invoke copy assignment.